File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/converters/ConverterDateInt.java
/*
 * Copyright (c) Kuba Szczodrzyński 2019-10-17.
 */

package pl.szczodrzynski.edziennik.data.db.converters;

import androidx.room.TypeConverter;

import pl.szczodrzynski.edziennik.utils.models.Date;

public class ConverterDateInt {

    @TypeConverter
    public static Date toDate(int value) {
        return Date.fromYmd(Integer.toString(value));
    }

    @TypeConverter
    public static int toInt(Date value) {
        return value == null ? 0 : value.getValue();
    }
}

File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/converter/ConverterDateInt.kt
/*
 * Copyright (c) Kuba Szczodrzyński 2019-10-17.
 */
package pl.szczodrzynski.edziennik.data.db.converter

import androidx.room.TypeConverter
import pl.szczodrzynski.edziennik.utils.models.Date

class ConverterDateInt {
    @TypeConverter
    fun toDate(value: Int): Date? = if (value == 0) null else Date.fromValue(value)

    @TypeConverter
    fun toInt(date: Date?): Int = date?.value ?: 0
}
